src/aind_metadata_mapper/bergamo/session.py

```diff
@@ -0,0 +1,548 @@
+            input_source = self.input_source
+
+        if os.path.isfile(input_source):
+            file_with_metadata = input_source
+        else:
+            file_with_metadata = self._get_si_file_from_dir(input_source)
+        # Not sure if a custom header was appended, but we can't use
+        # o=json.loads(reader.metadata()) directly
+        with ScanImageTiffReader(str(file_with_metadata)) as reader:
+            img_metadata = reader.metadata()
+            img_description = reader.description(0)
+            img_shape = reader.shape()
+        return RawImageInfo(
+            metadata=img_metadata,
+            description0=img_description,
+            shape=img_shape,
+        )
+
+    def _transform(self, extracted_source: RawImageInfo) -> Session:
+        """
+        Transforms the raw data extracted from the tif directory into a
+        Session object.
+        Parameters
+        ----------
+        extracted_source : RawImageInfo
+
+        Returns
+        -------
+        Session
+
+        """
+        siHeader = self._parse_raw_image_info(extracted_source)
+        photostim_groups = siHeader.metadata["json"]["RoiGroups"][
+            "photostimRoiGroups"
+        ]
+
+        data_stream = Stream(
+            stream_start_time=self.user_settings.stream_start_time,
+            stream_end_time=self.user_settings.stream_end_time,
+            stream_modalities=[Modality.POPHYS],
+            camera_names=list(self.user_settings.camera_names),
+            light_sources=[
+                Laser(
+                    name="Laser A",
+                    wavelength=920,
+                    wavelength_unit="nanometer",
+                    excitation_power=int(
+                        siHeader.metadata["hBeams"]["powers"][1:-1].split()[0]
+                    ),
+                    excitation_power_unit="percent",
+                ),
+            ],
+            detectors=[
+                Detector(
+                    name="PMT A",
+                    exposure_time=0.1,
+                    trigger_type="Internal",
+                ),
+            ],
+            ophys_fovs=[
+                FieldOfView(
+                    index=0,
+                    imaging_depth=150,
+                    targeted_structure="M1",
+                    fov_coordinate_ml=1.5,
+                    fov_coordinate_ap=1.5,
+                    fov_reference="Bregma",
+                    fov_width=int(
+                        siHeader.metadata["hRoiManager"]["pixelsPerLine"]
+                    ),
+                    fov_height=int(
+                        siHeader.metadata["hRoiManager"]["linesPerFrame"]
+                    ),
+                    magnification="16x",
+                    fov_scale_factor=float(
+                        siHeader.metadata["hRoiManager"]["scanZoomFactor"]
+                    ),
+                    frame_rate=float(
+                        siHeader.metadata["hRoiManager"]["scanFrameRate"]
```

Comment:
@jtyoung84 Can you ping @kayvondaie to know if these should be hardcoded or if they need to be parameters?

Reply:
If we end up needing to make this part of a client-run script because it's changing all of the time that's fine.